vscode/oso/client/src/index.ts

```diff
@@ -175,9 +185,11 @@ async function startClient(folder: WorkspaceFolder, context: ExtensionContext) {
   };
   const client = new LanguageClient(extensionName, serverOpts, clientOpts);
 
-  context.subscriptions.push(
-    client.onTelemetry(event => enqueueEvent(folder.uri, event))
+  const recordTelemetry = debounce(
+    event => recordEvent(folder.uri, event),
+    1_000
   );
+  context.subscriptions.push(client.onTelemetry(recordTelemetry));
```

Comment:
This debounced function is initialized in `startClient` so that we have a separate debounced function for each client (one client per workspace folder).